framework/src/meshmodifiers/BreakMeshByBlock.C

```diff
@@ -0,0 +1,170 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BreakMeshByBlock.h"
+#include "MooseMesh.h"
+
+registerMooseObject("MooseApp", BreakMeshByBlock);
+
+template <>
+InputParameters
+validParams<BreakMeshByBlock>()
+{
+  InputParameters params = validParams<BreakMeshByBlockBase>();
+  params.addClassDescription("Break boundaries based on the subdomains to which their sides are "
+                             "attached. Naming convention for the new boundaries will be the old "
+                             "boundary name plus \"_to_\" plus the subdomain name. At the moment"
+                             "this only works on REPLICATED mesh");
+  params.addParam<std::vector<BoundaryName>>(
+      "boundaries", "Boundaries to be broken. Default means to break all boundaries");
+  return params;
+}
+
+BreakMeshByBlock::BreakMeshByBlock(const InputParameters & parameters)
+  : BreakMeshByBlockBase(parameters)
+{
+}
+
+void
+BreakMeshByBlock::modify()
+{
+
+  // TODO remove when distributed MESH capabilities are implemented
+  _mesh_ptr->errorIfDistributedMesh("BreakMeshByBlock only works on a REPLICATED mesh");
+  checkInputParameter();
+
+  // save reference to mesh
+  MeshBase & mesh = _mesh_ptr->getMesh();
+
+  // initialize the node to elemen map
+  const std::map<dof_id_type, std::vector<dof_id_type>> & node_to_elem_map =
+      _mesh_ptr->nodeToElemMap();
```

Comment:
@friedmud Should we explicitly call `MooseMesh::update()` at the end of the main loop to reinit the `_node_to_element_map` or it is done automatically at the end of each mesh modifier?

Reply:
I'd have to double check, but I'm going to say don't call it. We try to minimize calls to reinit and prepare because they can get expensive so MOOSE generally does all of this when it's needed. If you run into a problem, let us know.

Reply:
I honestly don't know - can you go put a print in the nodeToElemMap() function (where it gets rebuilt) to make sure that it _does_ get rebuilt after your modifier?

I think it does - but always good to check

Reply:
@friedmud 
I added I print statement at the end of the `BreakMeshByBlock::modify` and one iniside `MooseMesh::update`

the result is  the following
```
MooseMesh::update() reset _node_to_elem_map
END OF BreakMeshByBlock
MooseMesh::update() reset _node_to_elem_map
```

So it gets resetted at some point

Reply:
👍